lib/spack/spack/ci/common.py

```diff
@@ -479,284 +770,221 @@ class SpackCIConfig:
     used by the CI generator(s).
     """
 
-    def __init__(self, ci_config):
+    def __init__(self, ci_config: Dict[str, Any]):
         """Given the information from the ci section of the config
         and the staged jobs, set up meta data needed for generating Spack
         CI IR.
         """
-
         self.ci_config = ci_config
-        self.named_jobs = ["any", "build", "copy", "cleanup", "noop", "reindex", "signing"]
-
-        self.ir = {
-            "jobs": {},
-            "rebuild-index": self.ci_config.get("rebuild-index", True),
-            "broken-specs-url": self.ci_config.get("broken-specs-url", None),
-            "broken-tests-packages": self.ci_config.get("broken-tests-packages", []),
-            "target": self.ci_config.get("target", "gitlab"),
+
+        self.broken_specs_url: Optional[str] = ci_config.get("broken-specs-url", None)
+        self.broken_tests: List[str] = ci_config.get("broken-tests-packages", [])
+        self.rebuild_index: bool = ci_config.get("rebuild-index", True)
+
+        self.pipeline_gen: List[Dict[str, Dict]] = ci_config.get("pipeline-gen", [])
+
+        # TODO: Move this to GitlabConfig?
+        #: Callable methods that perform variable substitutions on script lines
+        self.converter: Dict[CIScriptStage, Callable] = {}
+
+        # Prep for each possible job type
+        self.jobs: Dict[str, List[CIJobData]] = {}
+        for name in all_job_names:
+            self.jobs[name] = []
+
+        # Add each core job along with its defaults
+        for name in core_job_names:
+            tty.debug(f"TLD: SpackCIConfig: instantiating '{name}' job")
+            self.jobs[name] = [CIJobData(name)]
+
+    # Create jobs for all the pipeline specs
+    def init_pipeline_jobs(self, pipeline: PipelineDag, tests: Optional[bool] = False):
+        names = ["build"]
+        if tests:
+            names.append("test")
```

Comment:
This should always be on, but only initialize the `CIJobData` if a `test-job` is defined for a spec.